### ISSUE TITLE
Incomplete message

### ISSUE THREAD

**Issue by aniketpant1** [syslog-ng/syslog-ng#3733]
# syslog-ng
## Version of syslog-ng
3.30

## Platform
oracle linux

## Debug bundle

Syslog-NG DEBUg buNdle generator
Start environment detection
Linux-type FHS detected

Operating System Name: Linux

Redhat specific checks
Check package files integrity
package syslog-ng-premium-edition is not installed
package syslog-ng-premium-edition-client is not installed
package syslog-ng-premium-edition-compact is not installed
No syslog-ng RPM packages have been found!
list syslog-related packages

Start general info collection
System's full uname: Linux kaf1 4.18.0-147.el8.x86_64 #1 SMP Wed Dec 4 21:51:45 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
Getting network-interface information: Success
Getting network routes: Success
Getting DNS resolution-related information: Done
List all processes
Mount and disk free info collection

Start Syslog-specific info collection
Copy configuration files from /etc/syslog-ng
24 blocks
Copy SCL configuration files
236 blocks
Gathering PKI information... done.
SVpid: 1 SNGpid: 57129 Chpids:
Syslog-ng's exact version: Error opening plugin module; module='mod-java', error='libjvm.so: cannot open shared object file: No such file or directory'
syslog-ng 3 (3.31.2)
/usr/sbin/syslog-ng-ctl stats
/usr/sbin/syslog-ng-ctl query get *
/usr/sbin/syslog-ng-ctl show-license-info
/usr/sbin/syslog-ng-ctl credentials status
[2021-07-13T17:08:43.865119] WARNING: value references in filters should not use the '$' prefix, those are only needed in templates, removing automatically; value='$PROXIED_SRCIP', location='/etc/syslog-ng/conf.d/kafka_destination.conf:42:60'
[2021-07-13T17:08:43.865182] WARNING: value references in filters should not use the '$' prefix, those are only needed in templates, removing automatically; value='$PROXIED_SRCIP', location='/etc/syslog-ng/conf.d/kafka_destination.conf:43:60'
[2021-07-13T17:08:43.865209] WARNING: value references in filters should not use the '$' prefix, those are only needed in templates, removing automatically; value='$PROXIED_SRCIP', location='/etc/syslog-ng/conf.d/kafka_destination.conf:44:60'
[2021-07-13T17:08:43.865241] WARNING: value references in filters should not use the '$' prefix, those are only needed in templates, removing automatically; value='$PROXIED_SRCIP', location='/etc/syslog-ng/conf.d/kafka_destination.conf:45:59'
[2021-07-13T17:08:43.865263] WARNING: value references in filters should not use the '$' prefix, those are only needed in templates, removing automatically; value='$HOST_FROM', location='/etc/syslog-ng/conf.d/kafka_destination.conf:45:101'
32 blocks
Detecting init system: systemd detected...
Generating second batch of statistics
/usr/sbin/syslog-ng-ctl stats
/usr/sbin/syslog-ng-ctl query get *

Generating hashes... done.
Debug Bundle generation: Done.
tar: ./var/syslog-ng.ctl: socket ignored
Terminating live message watcher: /usr/sbin/syslog-ng-debun: line 293: 58001 Terminated              tail -f ${tmpdir}/syslog-ng.debun.txt 1>&3  (wd: ~)

# Issue
two cisco devices one uses rfc5424 and second one uses rfc3164.Some messages are not coming or message parts are breaking(in case of rfc3164)
## Failure

[2021-07-13T17:30:20.337628] WARNING: value references in filters should not use the '$' prefix, those are only needed in templates, removing automatically; value='$PROXIED_SRCIP', location='/etc/syslog-ng/conf.d/kafka_destination.conf:42:60'
[2021-07-13T17:30:20.337695] WARNING: value references in filters should not use the '$' prefix, those are only needed in templates, removing automatically; value='$PROXIED_SRCIP', location='/etc/syslog-ng/conf.d/kafka_destination.conf:43:60'
[2021-07-13T17:30:20.337728] WARNING: value references in filters should not use the '$' prefix, those are only needed in templates, removing automatically; value='$PROXIED_SRCIP', location='/etc/syslog-ng/conf.d/kafka_destination.conf:44:60'
[2021-07-13T17:30:20.337759] WARNING: value references in filters should not use the '$' prefix, those are only needed in templates, removing automatically; value='$PROXIED_SRCIP', location='/etc/syslog-ng/conf.d/kafka_destination.conf:45:59'
[2021-07-13T17:30:20.337783] WARNING: value references in filters should not use the '$' prefix, those are only needed in templates, removing automatically; value='$HOST_FROM', location='/etc/syslog-ng/conf.d/kafka_destination.conf:45:101'
[2021-07-13T17:30:20.342099] WARNING: With use-dns(no), dns-cache() will be forced to 'no' too!;
[2021-07-13T17:30:20.345951] Error binding socket; addr='AF_INET(0.0.0.0:1470)', error='Address already in use (98)'
[2021-07-13T17:30:20.345979] Error initializing message pipeline; plugin_name='network', location='/etc/syslog-ng/conf.d/kafka_destination.conf:11:6'

## Steps to reproduce

i want to capture cisco ise and cisco asa syslog device logs.The ise device uses rfc3164 and asa device uses rfc5424.I am using haproxy tool.This two devices sends logs to the syslog-ng servers(we setup 3 syslog-ng server) through haproxy load-balancer.
The cisco asa device logs is working fine but not in case of ise device although configuration file is same
```
source s_network_tcp {
     network(
     transport("proxied-tcp")
     port(1470)
);
};
#############################################

rewrite pfsense_tag {
set("pfsense",value("tags"));
};
rewrite vpn1_tag {
set("vpn1", value("tags"));
set("$PROXIED_SRCIP", value("HOST_FROM"));
};
rewrite vpn2_tag {
set("vpn2",value("tags"));
set("$PROXIED_SRCIP", value("HOST_FROM"));
};
rewrite vpn7_tag {
set("vpn7",value("tags"));
set("$PROXIED_SRCIP", value("HOST_FROM"));
};
rewrite ise_tag {
set("ise",value("tags"));
set("$PROXIED_SRCIP",value("HOST_FROM"));
};

############################################

filter host99{ host("xx.xx.xx.xx") or host("xx.xx.xx.xx") };

filter vpn1_host{match("xx.xx.xx.xx" value("$PROXIED_SRCIP"))};
filter vpn2_host{match("xx.xx.xx.xx" value("$PROXIED_SRCIP"))};
filter vpn7_host{match("xx.xx.xx.xx" value("$PROXIED_SRCIP"))};
filter ise_host{match("xx.xx.xx.xx" value("$PROXIED_SRCIP")) or host("xxxxx")};
#############################################
destination pfsense_kafka
{
kafka (
bootstrap-servers("kafka:9092")
topic("pfsense")
 message("$(format-json time=$ISODATE tags host=$HOST message='${MSGHDR}${MSG}')")
);
};

destination vpn1_kafka
{
kafka (
bootstrap-servers("lkafka:9092")
topic("vpn1")
message("$(format-json time=$ISODATE tags proxied_srcip=$PROXIED_SRCIP host=$HOST message='${MSGHDR}${MSG}')")
);
};

destination vpn2_kafka
{
kafka (
bootstrap-servers("kafka:9092")
topic("vpn2")
message("$(format-json time=$ISODATE tags proxied_srcip=$PROXIED_SRCIP host=$HOST message='${MSGHDR}${MSG}')")
);
};

destination vpn7_kafka
{
kafka (
bootstrap-servers("kafka:9092")
topic("vpn7")
message("$(format-json time=$ISODATE tags proxied_srcip=$PROXIED_SRCIP host=$HOST message='${MSGHDR}${MSG}')")
);
};

destination ise_kafka
{
kafka (
bootstrap-servers("kafka:9092")
topic("ise")
message("$(format-json time=$ISODATE tags proxied_srcip=$PROXIED_SRCIP  host=$HOST message='${MSGHDR}${MSG}')")
);
};

############################################

log{ source(s_network_tcp);filter(vpn1_host);rewrite(vpn1_tag); destination(vpn1_kafka);};
log{ source(s_network_tcp);filter(vpn2_host);rewrite(vpn2_tag); destination(vpn2_kafka);};
log{ source(s_network_tcp);filter(vpn7_host);rewrite(vpn7_tag); destination(vpn7_kafka);};
log{ source(s_network_tcp);filter(ise_host);rewrite(ise_tag); destination(ise_kafka);};
```
yes  things i'venoticed that ise device has hostname but in log entry we see only ip in the field of host.In cisco asa device the host field has the ip of haproxy but in case of ise we see sometime ip of ise device instead of hostname and sometime ip of haproxy
## Configuration
**/etc/syslog-ng/syslog.conf**
```

@version: 3.30
@include "scl.conf"
#@module mod-java
#@define kafka-implementation kafka-c
# syslog-ng configuration file.
#
# This should behave pretty much like the original syslog on RedHat. But
# it could be configured a lot smarter.
#
# See syslog-ng(8) and syslog-ng.conf(5) for more information.
#
# Note: it also sources additional configuration files (*.conf)
#       located in /etc/syslog-ng/conf.d/

options {
    flush_lines (0);
    time_reopen (10);
    log_fifo_size (1000);
    chain_hostnames (off);
    use_dns (no);
    use_fqdn (no);
    create_dirs (no);
    keep_hostname (yes);
};


# Source additional configuration files (.conf extension only)
@include "/etc/syslog-ng/conf.d/*.conf"
# vim:ft=syslog-ng:ai:si:ts=4:sw=4:et:
```

**Comment by szemere** [comment]
Hello @aniketpant1,

Thank You for the detailed description. I have noticed two errors in your debun output, I would recommend to fix them:

- `WARNING: value references in filters should not use the '$' prefix, those are only needed in templates, removing automatically; value='$PROXIED_SRCIP', location='/etc/syslog-ng/conf.d/kafka_destination.conf:42:60'`
The problem is here:
`
rewrite vpn1_tag {
set("vpn1", value("tags"));
set("$PROXIED_SRCIP", value("HOST_FROM"));
};`

- `Error binding socket; addr='AF_INET(0.0.0.0:1470)', error='Address already in use (98)'`
Most likely there is an another running Syslog-ng instance. (But it is possible that there is an another program which uses that port.)


About the CISCO issues. If I understood you correctly, there are two problems in parallel.
1. Not receiving certain messages.
2. There might be a parser issue in Syslog-ng. (The actual output is different from what you expect.)
 

About the first problem: We have to decide if the messages are actually reaching Syslog-ng, or somehow Syslog-ng drops them by error. The first part is easy. As a test, you can start an another program (i.e. netcat) to receive messages from the network.

AFTER you clarified that those messages are indeed reaching your application, than we can investigate Syslog-ng. As a first step, Syslog-ng is writing a debug log from every received message. Starting with the text: "Incoming log entry" in the debug logs.
If you can see the debug message than it is most likely that those messages are dropped by a filter rule.


About the parser issue: To confirm this, we would need a problematic message from the CISCO device. There are many good ways to reproduce an issue with a fixed message. One example:

- put one problematic message into a file (mask any sensitive information, make sure to not change the syntax of the message)
- read that file with flags(no-parse)
- put any parser in the pipeline
- use the questionable field in an output template

```
source s_problematic_message {
  file("/tmp/problematic.message"
    flags(no-parse)
  );
};

parser p_syslog {
  syslog-parser(flags(syslog-protocol));
};

destination d_output {
  file("/tmp/output.message"
    template("host: $HOST ; msg: $MSG\n")
  );
};

log {
  source(s_problematic_message);
  parser(p_syslog);
  destination(d_output);
};
```

If the issue is reproducible with the file source, than you can share your file with us. We can use that file to debug the parser code and fix the problem.

Best regards,
Laci

**Comment by aniketpant1** [comment]
Hey @szemere i am using 3 syslog-ng server basically my ciso ise send logs to the all syslog-ng server through haproxy load-balancer.I made some changes generally when any log comes to the syslog-ng and then output to the kafka.The message format is `time:xxxx , tags:xxx , proxied_scrip:cisco_ise_ip , message:xxxxxxxxxxxx , host:haproxy_ip`
in host field we sometime sees cisco_ise_ip and sometimes haproxy but we as we specified in above configuration we are filtering the message based on cisco_ise_ip

For testing purpose i made some changes in filter function i am using haproxy and load_balancer_ip as i told you that if a message travel through load balancer then in host field we can see only load-balancer-ip and in proxied_scrip we can see ip of original source
```
source s_network_tcp {
     network(
     transport("proxied-tcp")
     port(1470)
);
};
rewrite ise_tag {
set("ise",value("tags"));
set("$PROXIED_SRCIP",value("HOST_FROM"));
};
filter ise_host_test{match("xx.xx.xx..44" value("$PROXIED_SRCIP")) and match("xx.xx.xx.xx.249" value ("$HOST"))};
destination test_ise_kafka
{
kafka (
bootstrap-servers("xx.xx.xx.xx:9092")
topic("test_ise")
message("$(format-json time=$ISODATE tags proxied_srcip=$PROXIED_SRCIP  host=$HOST message='${MSGHDR}${MSG}')")
);
};

log{ source(s_network_tcp);filter(ise_host_test);rewrite(ise_tag); destination(test_ise_kafka);};
```
This is the output i got
```
{"time":"2021-07-20T14:29:00+05:30","tags":"ise","proxied_srcip":"xx.xx.xx.44","message":"E=691 R=0 C=18B0B28E7559310A16A62B5BA856B2E5 V=3; },","host":"xx.xx.xx.249"}
```
So i am facing 2 problems

1.  some messages are not coming
2. messages are coming but it is incomplete

Answer of 1 question is that i run tcpdump on haproxy server to so that i can investigate is that particular message is reaching haproxy server because in syslog-ng server i didn't see and in haproxy server tcpdump it also didn't shown there

**Comment by szemere** [comment]
Hello @aniketpant1 ,

thank you for the configuration snippet. One note about your config:

 There is a rewrite rule: `set("$PROXIED_SRCIP",value("HOST_FROM"));`
 But I failed to see where this macro is used later in your configuration. In the mentioned blog post, the output is a JSON nv-pairs, which contians both the HOST_FROM and the HOST fields and they have a different meaning in that context.


I checked the source code to clarify some things. Here are my findings:

- There are some global options ("use-dns, keep-hosname, use-fqdn") which can alter the HOST field in case of network sources. Please make sure to also check those values. I don't know if you don't have any global options or just missed them from the snippet above.
- The 'proxied-tcp' transport do not touch the HOST and HOST_FROM macros. Only the fields prefixed with PROXIED_
- Those proxied macros are copied as is. So the global "use-dns, keep-hosname, use-fqdn" options do not alters them.

---

For question 1:

> Answer of 1 question is that i run tcpdump on haproxy server to so that i can investigate is that particular message is reaching haproxy server because in syslog-ng server i didn't see and in haproxy server tcpdump it also didn't shown there

Just to clarify things: your setup is `cisco device -> haproxy -> syslog-ng`, and those logs are not reaching the HAPROXY server? Before Syslog-ng.

---

One small note: Maybe I misundertood the term "incomplete". I think it would clarify things if you can attach:

- One (or more) example message. (It don't have to be from your network, if it is containing sensitive information. But in that case you have to work out the differences.)
- One WORKING configuration example. Which I can load into the latest version of Syslog-ng directly.
- The expected result.


Why am I asking for these informations? There are many components in your setup outside of Syslog-ng. Since we do not have a chance to fully know your setup. (Exact type and configuration of your devices, proxy server, kafka server, etc...) The best thing we can do is to focus on Syslog-ng.
Debugging your setup as is, is the area of a system integrator, and can not be done in an issue ticket.


Br,
Laci

**Comment by aniketpant1** [comment]
@szemere thanks for reply
this is the output i can show you this is the message that is coming in tcpdump and also in syslog-ng but in syslog-ng the message is not complete
```
<189>Jul 27 11:49:31 host_name CISE_Failed_Attempts 0000642211 4 3  Response={RadiusPacketType=AccessReject; AuthenticationResult=Failed; MS-CHAP-Error=.E=691 R=0 C=C95E30D7B53CDF5A7FF8D1DD5420C6FF V=3; }
```
and the message we got in syslog-ng,sorry i have the old data but the issue is still same
```
{"time":"2021-07-24T09:50:07+05:30","tags":"ise","proxied_srcip":"source_ip","message":"CISE_Failed_Attempts 0000193089 4 3  Device Type=Device Type#All Device Types, IPSEC=IPSEC#Is IPSEC Device, EnableFlag=Enabled, Response={RadiusPacketType=AccessReject; AuthenticationResult=Failed; MS-CHAP-Error=","host":"source_hostname"}
```
the message gets break down
```
{"time":"2021-07-27T16:10:30+05:30","tags":"ise","proxied_srcip":"source_ip","message":"E=691 R=0 C=938F6696B31A84988D26D3D08A65F5AF V=3; },","host":"load_balancer_ip"}
```
The term 'incomplete' here has 2 meaning 1.message is coming but it is breaking that i shown you 2. message is not reaching but this is the issue of haproxy

**Comment by szemere** [comment]
Hello @aniketpant1,

I used the following example message:
```
<189>Jul 27 11:49:31 host_name CISE_Failed_Attempts 0000642211 4 3  Response={RadiusPacketType=AccessReject; AuthenticationResult=Failed; MS-CHAP-Error=.E=691 R=0 C=C95E30D7B53CDF5A7FF8D1DD5420C6FF V=3; }
```

your example message do not contains HAPROXY headers. So I assume they are not from your actual setup.

PLEASE provide **one message** and **one working configuration** which will  **reproduce your problem**. Otherwise I assume this is **NOT** an issue with Syslog-ng.

What do I mean? Your message produces the following error with the network source from your previous post:

```
[2021-07-27T12:17:44.899173] Initializing PROXY protocol source driver; driver='0x5607aae83b10'
[2021-07-27T12:17:44.899223] Syslog connection accepted; fd='13', client='AF_INET(127.0.0.1:48834)', local='AF_INET(0.0.0.0:1470)'
[2021-07-27T12:17:44.899365] PROXY proto header length exceeds max. length defined by specification; length='204', header='<189>Jul 27 11:49:31 host_name CISE_Failed_Attempts 0000642211 4 3  Response={RadiusPacketType=AccessReject; AuthenticationResult=Failed; MS-CHAP-Error=.E=691 R=0 C=C95E30D7B53CDF5A7FF8D1DD5420C6FF V=3; }\x0a'
[2021-07-27T12:17:44.899380] PROXY protocol header received; line='<189>Jul 27 11:49:31 host_name CISE_Failed_Attempts 0000642211 4 3  Response={RadiusPacketType=AccessReject; AuthenticationResult=Failed; MS-CHAP-Error=.E=691 R=0 C=C95E30D7B53CDF5A7FF8D1DD5420C6FF V=3; }'
[2021-07-27T12:17:44.899383] Error parsing PROXY protocol header;
[2021-07-27T12:17:44.899406] Syslog connection closed; fd='13', client='AF_INET(127.0.0.1:48834)', local='AF_INET(0.0.0.0:1470)'
[2021-07-27T12:17:44.899418] Freeing PROXY protocol source driver; driver='0x5607aae83b10'
[2021-07-27T12:17:44.899443] Closing log transport fd; fd='13'
```


If I remove the `transport("proxied-tcp")` part, the remaining message will be parsed as follow:

```
[2021-07-27T12:33:26.965797] Syslog connection accepted; fd='13', client='AF_INET(127.0.0.1:48954)', local='AF_INET(0.0.0.0:1470)'
[2021-07-27T12:33:26.966232] Incoming log entry; line='<189>Jul 27 11:49:31 host_name CISE_Failed_Attempts 0000642211 4 3  Response={RadiusPacketType=AccessReject; AuthenticationResult=Failed; MS-CHAP-Error=.E=691 R=0 C=C95E30D7B53CDF5A7FF8D1DD5420C6FF V=3; }'
[2021-07-27T12:33:26.966288] Initial message parsing follows;
[2021-07-27T12:33:26.966374] Setting value; name='PROGRAM', value='CISE_Failed_Attempts', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966400] Setting value; name='LEGACY_MSGHDR', value='CISE_Failed_Attempts ', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966413] Setting value; name='HOST', value='host_name', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966436] Setting value; name='MESSAGE', value='0000642211 4 3  Response={RadiusPacketType=AccessReject; AuthenticationResult=Failed; MS-CHAP-Error=.E=691 R=0 C=C95E30D7B53CDF5A7FF8D1DD5420C6FF V=3; }', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966475] >>>>>> Source side message processing begin; instance='tcp,127.0.0.1', location='/tmp/aniketpant1.conf:29:6', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966501] Setting value; name='HOST_FROM', value='127.0.0.1', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966530] Setting value; name='SOURCE', value='s_network_tcp', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966549] >>>>>> rewrite rule evaluation begin; rule='vpn1_tag', location='/tmp/aniketpant1.conf:39:1', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966581] Setting value; name='tags', value='vpn1', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966599] <<<<<< rewrite rule evaluation finished; rule='vpn1_tag', location='/tmp/aniketpant1.conf:39:1', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966616] >>>>>> rewrite rule evaluation begin; rule='vpn1_tag', location='/tmp/aniketpant1.conf:40:1', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966637] Setting value; name='HOST_FROM', value='', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966653] <<<<<< rewrite rule evaluation finished; rule='vpn1_tag', location='/tmp/aniketpant1.conf:40:1', msg='0x7f3398015b70'
[2021-07-27T12:33:26.966870] Initializing destination file writer; template='/tmp/aniketpantl1.conf', filename='/tmp/aniketpantl1.conf'
[2021-07-27T12:33:26.966981] affile_open_file; path='/tmp/aniketpantl1.conf', fd='15'
[2021-07-27T12:33:26.967075] <<<<<< Source side message processing finish; instance='tcp,127.0.0.1', location='/tmp/aniketpant1.conf:29:6', msg='0x7f3398015b70'
```

From the debug logs I can see that the `HOST` value is parsed correctly from the message.
```
[2021-07-27T12:33:26.966413] Setting value; name='HOST', value='host_name', msg='0x7f3398015b70'
```

Br,
Laci

**Comment by aniketpant1** [comment]
I am showing the output of `syslog-ng -Fedtv` 
This what happens when that particular message received by syslog-ng
```
[2021-07-27T18:31:36.905050] Incoming log entry; line='<189>Jul 27 18:31:36 ised-vpn CISE_Failed_Attempts 0000764569 4 3  Response={RadiusPacketType=AccessReject; AuthenticationResult=Failed; MS-CHAP-Error='
[2021-07-27T18:31:36.905053] Initial message parsing follows;
[2021-07-27T18:31:36.905056] Setting value; name='PROGRAM', value='CISE_Failed_Attempts', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905060] Setting value; name='LEGACY_MSGHDR', value='CISE_Failed_Attempts ', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905063] Setting value; name='HOST', value='ised-vpn', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905067] Setting value; name='MESSAGE', value='0000764569 4 3  Response={RadiusPacketType=AccessReject; AuthenticationResult=Failed; MS-CHAP-Error=', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905070] Setting value; name='PROXIED_SRCIP', value='source_ip', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905074] Setting value; name='PROXIED_DSTIP', value='load_balancer-ip', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905077] Setting value; name='PROXIED_SRCPORT', value='22314', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905080] Setting value; name='PROXIED_DSTPORT', value='1468', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905083] Setting value; name='PROXIED_IP_VERSION', value='4', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905088] >>>>>> Source side message processing begin; instance='proxied-tcp,load_balancer-ip', location='/etc/syslog-ng/conf.d/rsyslog_input.conf:13:6', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905091] Setting value; name='HOST_FROM', value='load_balancer-ip', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905095] Setting value; name='SOURCE', value='s_network_tcp', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905098] >>>>>> filter rule evaluation begin; rule='src4', location='/etc/syslog-ng/conf.d/rsyslog_input.conf:28:13', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905103] match() evaluation started; input='source_ip', pattern='source_ip', value='PROXIED_SRCIP', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905106] match() evaluation started; input='CISE_Failed_Attempts', pattern='CISE_Failed_Attempts*', value='PROGRAM', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905110] >>>>>> rewrite rule evaluation begin; rule='r_rewrite_src4', location='/etc/syslog-ng/conf.d/rsyslog_input.conf:33:1', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905113] Setting value; name='tags', value='ise', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905117] <<<<<< rewrite rule evaluation finished; rule='r_rewrite_src4', location='/etc/syslog-ng/conf.d/rsyslog_input.conf:33:1', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905120] >>>>>> rewrite rule evaluation begin; rule='r_rewrite_src4', location='/etc/syslog-ng/conf.d/rsyslog_input.conf:34:1', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905123] Setting value; name='HOST_FROM', value='source_ip', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905127] <<<<<< rewrite rule evaluation finished; rule='r_rewrite_src4', location='/etc/syslog-ng/conf.d/rsyslog_input.conf:34:1', msg='0x7fa35803e6b0'
[2021-07-27T18:31:36.905131] <<<<<< filter rule evaluation result; result='MATCH - Forwarding message to the next LogPipe', rule='src4', location='/etc/syslog-ng/conf.d/rsyslog_input.conf:28:13', msg='0x7fa35803e6b0'

[2021-07-27T18:31:36.905142] Incoming log entry; line='E=691 R=0 C=C1C1F0594A2AA531C950C00BD70B5F3D V=3; },'
[2021-07-27T18:31:36.905144] Initial message parsing follows;
[2021-07-27T18:31:36.905156] Setting value; name='PROGRAM', value='E=691', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905159] Setting value; name='LEGACY_MSGHDR', value='E=691 ', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905162] Setting value; name='MESSAGE', value='R=0 C=C1C1F0594A2AA531C950C00BD70B5F3D V=3; },', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905166] Setting value; name='PROXIED_SRCIP', value='source_ip', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905169] Setting value; name='PROXIED_DSTIP', value='load_balancer-ip', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905172] Setting value; name='PROXIED_SRCPORT', value='22314', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905176] Setting value; name='PROXIED_DSTPORT', value='1468', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905179] Setting value; name='PROXIED_IP_VERSION', value='4', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905183] >>>>>> Source side message processing begin; instance='proxied-tcp,load_balancer-ip', location='/etc/syslog-ng/conf.d/rsyslog_input.conf:13:6', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905186] Setting value; name='HOST_FROM', value='load_balancer-ip', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905190] Setting value; name='HOST', value='load_balancer-ip', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905193] Setting value; name='SOURCE', value='s_network_tcp', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905197] >>>>>> filter rule evaluation begin; rule='src4', location='/etc/syslog-ng/conf.d/rsyslog_input.conf:28:13', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905200] match() evaluation started; input='source_ip', pattern='source_ip', value='PROXIED_SRCIP', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905203] match() evaluation started; input='E=691', pattern='CISE_Failed_Attempts*', value='PROGRAM', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905207] <<<<<< filter rule evaluation result; result='UNMATCHED - Dropping message from LogPipe', rule='src4', location='/etc/syslog-ng/conf.d/rsyslog_input.conf:28:13', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905211] <<<<<< Source side message processing finish; instance='proxied-tcp,load_balancer-ip', location='/etc/syslog-ng/conf.d/rsyslog_input.conf:13:6', msg='0x7fa358009c60'
[2021-07-27T18:31:36.905470] kafka: message published; topic='ise_tcp', key='NULL', message='{"time":"2021-07-27T18:31:36+05:30","tags":"ise","proxied_srcip":"10.1.145.44","message":"CISE_Failed_Attempts 0000764569 4 3  Response={RadiusPacketType=AccessReject; AuthenticationResult=Failed; MS-CHAP-Error=","host":"ised-vpn"}', driver='ise_kafka_tcp#0', location='#buffer:4:5'
```
this is original message when it reach to syslog-ng through load-balancer(haproxy)
```
<189>Jul 27 11:49:31 host_name CISE_Failed_Attempts 0000642211 4 3  Response={RadiusPacketType=AccessReject; AuthenticationResult=Failed; MS-CHAP-Error=.E=691 R=0 C=C95E30D7B53CDF5A7FF8D1DD5420C6FF V=3; }
```
in the above debug log the you can see UNMATCHED IP `E=691 R=0 C=C95E30D7B53CDF5A7FF8D1DD5420C6FF V=3; }`

**Comment by aniketpant1** [comment]
hey @szemere i think there is some issue syslog-ng cannot process '.'
```
<189>Jul 28 15:33:27 hostname CISE_Failed_Attempts 0000927969 4 3  EnableFlag=Enabled, Response={RadiusPacketType=AccessReject; AuthenticationResult=ChangePassword; MS-CHAP-Error=.E=648 R=0 C=D13CA38FE73B7D2D43BAA3AA3E228998 V=3; },
```

**Comment by aniketpant1** [comment]
Can anyone please help me to get me out of this issue.

**Comment by MrAnno** [comment]
Duplicate of #3749.
